src/backend/columnar/columnar_customscan.c

```diff
@@ -1316,6 +1320,66 @@ ColumnarScanPath_PlanCustomPath(PlannerInfo *root,
 }
 
 
+/*
+ * ReparameterizeMutator changes all varnos referencing the topmost parent of
+ * child_rel to instead reference child_rel directly.
+ */
+static Node *
+ReparameterizeMutator(Node *node, RelOptInfo *child_rel)
+{
+	if (node == NULL)
+	{
+		return NULL;
+	}
+	if (IsA(node, Var))
+	{
+		Var *var = castNode(Var, node);
+		if (bms_is_member(var->varno, child_rel->top_parent_relids))
+		{
+			var = copyObject(var);
+			var->varno = child_rel->relid;
```

Comment:
I was thinking what happens if we drop a column of the partition relation, but I see that postgres already doesn't allow doing that.

Reply:
To be clear, `varno` here is the relid, not the same as `varattno`.

Reply:
yeah, I was thinking that we are setting varno so that it points to child rel, but also thought that could it be the case there is a mismatch between varattno due to dropped columns.
then I realized that we cannot even drop inherited columns